From ae470335a234e336c3d0337f065e26b29751e668 Mon Sep 17 00:00:00 2001
From: Allan Odgaard <git@abetarda.com>
Date: Tue, 12 Jun 2018 00:01:39 +0200
Subject: [PATCH] =?UTF-8?q?Remove=20dependency=20on=20=E2=80=98ns=E2=80=99?=
 =?UTF-8?q?=20framework?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Since the ‘ns’ framework depends on OakFoundation this was a circular dependency and only sort of worked because ‘ns’ was not listed as a link target.

Upstream-Status: Backport [https://github.com/textmate/textmate/commit/ae470335a234e336c3d0337f065e26b29751e668.patch]
---
 Frameworks/OakFoundation/src/OakFoundation.mm | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git ./Frameworks/OakFoundation/src/OakFoundation.mm ./Frameworks/OakFoundation/src/OakFoundation.mm
index 0459943e2..64dba5cc8 100644
--- ./Frameworks/OakFoundation/src/OakFoundation.mm
+++ ./Frameworks/OakFoundation/src/OakFoundation.mm
@@ -1,12 +1,11 @@
 #import "OakFoundation.h"
 #import "NSString Additions.h"
-#import <ns/ns.h>
 
 std::string OakMoveToTrash (std::string const& path)
 {
 	NSURL* resultingItemURL;
 	if([[NSFileManager defaultManager] trashItemAtURL:[NSURL fileURLWithPath:[NSString stringWithCxxString:path]] resultingItemURL:&resultingItemURL error:nil])
-			return to_s([resultingItemURL path]);
+			return resultingItemURL.filePathURL.path.fileSystemRepresentation ?: NULL_STR;
 	else	return NULL_STR;
 }
 
